src/server/model_db.ts

```diff
@@ -0,0 +1,94 @@
+import * as tf from '@tensorflow/tfjs';
+import * as fs from 'fs';
+import * as path from 'path';
+
+const DEFAULT_MIN_UPDATES = 10;
+
+function getLatestId(dir: string): string {
+  return fs.readdirSync(dir).reduce((acc, val) => {
+    if (val.endsWith('.json') && val.slice(0, -5) > acc) {
+      return val.slice(0, -5);
+    } else {
+      return acc;
+    }
+  }, '0');
+}
+
+function generateNewId() {
+  return new Date().getTime().toString();
+}
+
+function readJSON(path: string): any {
+  return JSON.parse(fs.readFileSync(path).toString());
+}
+
+export class ModelDB {
+  dataDir: string;
+  modelId: string;
+  updating: boolean;
+  minUpdates: number;
+
+  constructor(dataDir: string, currentModel?: string, minUpdates?: number) {
+    this.dataDir = dataDir;
+    this.modelId = currentModel || getLatestId(dataDir);
+    this.updating = false;
+    this.minUpdates = minUpdates || DEFAULT_MIN_UPDATES;
+  }
+
+  updateFiles(): string[] {
+    return fs.readdirSync(path.join(this.dataDir, this.modelId));
+  }
+
+  currentVars(): tf.Tensor[] {
+    const file = path.join(this.dataDir, this.modelId + '.json');
+    const json = readJSON(file);
+    return json['vars'].map((v: {values: number[], shape: number[]}) => {
+      return tf.tensor(v.values, v.shape);
+    });
+  }
+
+  possiblyUpdate() {
+    if (this.updateFiles().length < this.minUpdates || this.updating) {
+      return;
+    }
+    this.updating = true;
+    this.update();
+    this.updating = false;
+  }
```

Comment:
This is probably not the right way to do concurrency control, but if node is single-threaded it might work. If we have multiple processes it definitely won't, though, and we might need to use some kind of file-based mutex library.

Reply:
Node is single threaded, but it might be better to return a Promise that resolves when updating is done.